Adds the Berserker Mask to science roles for 15 TCs

PKPenguin321 opened the pull request:
[webm here](https://puu.sh/uqj8C/57d21300ee.webm) <- it's a little outdated, doesn't have the third reagent button and puts way too many stimulants into me. also doesnt show off fancy sounds!

The sprites are all codersprites (but the onmob is really good I'm proud of it) but the only one that I feel is really bad and in need of change is the onground sprite for the mask, if anybody wants to tackle it:
![baneonground](https://cloud.githubusercontent.com/assets/8345184/23494013/d713ce2c-fec4-11e6-97f1-e7648ce49008.png)


- Role restricted traitor item, only available to Research Director, Scientists, Roboticists. Costs 15 TCs
- Gives you the Berserker "martial art" (it uses the martial arts system but isn't really one) while worn in the mask slot
- Gives you four action buttons:
- Backbreak, aggressive grab somebody then hit the button to lift them up and drop them on your knee for massive damage
- Cuffbreak, basically an infinite freedom implant that destroys the cuffs
- Stimulants, which injects you with some handy combat chems. Shares a 1.5 minute cool down with medicines. Has 1.75 minutes of cooldown.
- Medicines, injects you with some handy healing chems. Shares a cool down with Stimulants. Has 1.5 minutes of cooldown.
- Growth Hormone, which injects you with growth serum. Shares a cool down with the other two. Has 15 seconds of cooldown.
- Pulling the mask off does a little stun and adds a teeny bit of anacea (the chem that purges medicines and deals toxin)

Known issue: As far as I can tell the uplink entry doesn't actually appear, I think the restricted_roles var might not work with multiple jobs (or I entered them wrong)

PKPenguin321 committed:
adds the berserker mask
doesnt compile because of addtimer meme

PKPenguin321 committed:
further tweaks and fixes

PKPenguin321 commented on tgstation.dme:
```diff
@@ -14,7 +14,7 @@
 // BEGIN_INCLUDE
 #include "_maps\__MAP_DEFINES.dm"
-#include "_maps\tgstation2.dm"
+#include "_maps\runtimestation.dm"
```
Whoops

GunHog commented:
## HEAVY BREATHING

coiax commented:
Is this a reference to something? I'm otherwise not getting why scientists can become Mad Scientists.

PKPenguin321 commented:
Yeah I've left hints everywhere the biggest tell is probably the mask on-mob sprites and code comments

Robustin commented:
How long do the stimulants stay in your system? I'm too lazy to look up metabolization rates.

Also references memes aside this doesn't really seem fitting for scientists.

PKPenguin321 commented:
Yeah the idea is that it was gonna be an augment for roboticists but I got lazy and didn't feel like fixing autoimplanters, and since it was less robotics-y I figured why not let all of science enjoy it

I might change that if I stop being lazy

PKPenguin321 commented:
Also the stimulants stay for I think like 2 minutes ish, I could reduce them or increase their cooldown if you feel it's needed

Niknakflak commented:
what is it a reference too, I dont get it therefore it just seems weird

calzilla1 commented:
Rename posidens wrath and give it to the assistant

ExcessiveUseOfCobblestone commented:
if I pull that mask off will you die?

RandomMarine commented:
Why would you shoot a spaceman before throwing him out of an airlock?

ExcessiveUseOfCobblestone commented:
The flight plan I just filed with Centcomm lists me, my men, Dr. Pavel here but only one of you! First one to talk gets to stay on my station!

Robustin commented:
If I heard that right you get 2 minutes of stimulants on a 90 seconds cooldown.

Thats a permanent stim pack son. 

Stimulants are really, really strong.

PKPenguin321 commented:
Now that you mention it yeah

Will tweak

Metallusions commented:
Yeah, you should probably lower the drug dosages to 4u.

ShadowDeath6 commented:
made me chuckle

InsaneHyena commented:
The thing that bothers me the most - why science?

PKPenguin321 commented:
>the idea is that it was gonna be an augment for roboticists but I got lazy and didn't feel like fixing autoimplanters, and since it was less robotics-y I figured why not let all of science enjoy it

Kromgar commented:
So... this increases your size too right?

HES A BIG GUY

PKPenguin321 commented:
i could put growth serum in it...

lzimann commented on code/datums/martial/berserker.dm:
```diff
@@ -0,0 +1,130 @@
+/datum/martial_art/berserker
+	name = "Berserker"
+	var/datum/action/berserker/backbreak/backbreak = new/datum/action/berserker/backbreak()
```
You don't need to specify the type with new if you already typecasted the variable.

PKPenguin321 replied:
so just `new()` on the right of the `=` ?

lzimann replied:
Yea

PKPenguin321 commented:
Also @Xhuis you made the role-restricted antag items thing, right? Can you check and see what I did wrong to cause the issue I made in the OP?

PKPenguin321 committed:
fixes and tweaks
updated onground mask sprite with an outline
added variable cooldowns to synthesizer abilities
added growth hormone ability
added sounds to some abilities
shortened paths when generating the action buttons

PKPenguin321 commented:
Check commit details on the `fixes and tweaks` commit for updates! I've addressed Robutin's feedback as well as added a growth hormone button and other stuff.

Repukan commented:
A counter should increase everytime you use each of the chems. When the mask is taken off, this invisible counter should apply damage equal to the amount in this invisible counter accumulated at once.

optimumtact commented:
If you're just going to make antag items for murder that don't even really match the job theme as job specific antags items we'll just remove the framework

:-1: from me

AnturK commented on code/datums/martial/berserker.dm:
```diff
@@ -0,0 +1,146 @@
+	var/chemCD = FALSE
+
+/datum/action/berserker/backbreak
+	name = "Backbreak - Only usable while aggressively grabbing someone. You lift them up and break their back over your knee."
```
Tooltips already grab description no need to cram it all in the name.

optimumtact commented:
For the creatively bankrupt among you
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=32&start=2250#p262556

anconfuzedrock commented:
make this rd only and have you stuff all nearby immovable rods into a planter

Xhuis commented:
As far as I can tell it should be showing up for those three roles, @PKPenguin321. Not sure why you're having trouble.

Shapsy commented:
So it's a 15TC item that gives you stimulants on demand, gives you healing on demand, gives you an infinite freedom implant on demand, and gives you a powerful attack as well, and you give it to science because "why not"? This is just Mary Sue: the item

MMMiracles commented:
Bane memes don't really fit. It'd make sense if it was some sort of temporary hulk-like strength akin to something like nerd rage.

Enact your revenge against those who dare moonlight in your department.

FantasticFwoosh commented:
Needs to be thrown to the art-guys, concept codersprite ok but currently that thing is hideous

> Nobody cared who was until i made a thread about moonlighting

Now this with a traitor power-arm via TC sharing is pretty dope strong (does it also function as internals?)

> I do have balance concern issues with this object in that beserker style only has two moves and the rest of the said perks of the object are object derived (chems) the creep here is that the object fufills two implants at once ruining TC efficiency (why buy two when you can get this item)

Trying too hard **to be** the reference rather than actually bringing the idea into the scope of the game with a nod to the source material

> Suggestion to improve - Use the xenomorph plasma counter and instead turn the mouthkit into a electropack like item (mouthkit seals into place with use) that relies on plasmaman canisters to fuel your abilities with a small amount of toxins leakage. Give a generic bulllish combat move set (blindly charging like a alien leap in slow mo, knocking over people) and keep special abilities like cuff snappping to rely on plasma, if you're all spent you can't do anything and R&D can research more canisters

Grazyn commented:
Doesn't fit science, should probably be assistant only, Bane and his goons lived in the sewers (maint), and were mostly miserable people taking revenge on their betters.

PKPenguin321 commented:
Yeah you're right I should fix the augment and NODROP aspect and tighten it to just roboticist so it's got an actual reason to be role restricted

Would it be cooler if you could tag it, then forcefully augment somebody to turn them into a brainless brute who's loyal to you? Then you could give them chainsaw arms and create the ultimate warrior or whatever

ExcessiveUseOfCobblestone commented:
What is the connection between the item and the role[s] it is/are restricted to?

KorPhaeron commented:
I agree with @optimumtact 

We already have two martial arts in the uplink, why do we need a third?

RandomMarine commented:
Do what @FantasticFwoosh said. Make it so that it requires plasma gas to utilize it. Better fits the reference and justifies the science role restriction.

PKPenguin321 commented:
>We already have two martial arts in the uplink, why do we need a third?

Like I said in the OP this isn't really a martial art, just takes advantage of the system because it's got very clean code. I could tie the abilities to the mask itself if I had to.

InsaneHyena commented:
The mask just doesn't fit science as it is right now. For a job-specific item for science, it has zero ties to science.

GunHog commented:
The emagged Secborg unlocker item some random player suggested would probably fit RD/Robotics more (but not Scientists). Maybe even a choice of SyndieBorg? These role-items need to be good enough to be chosen over unrestricted muderbone weapons, such as the dual e-sword or bombs.

This could possibly go to the Medical guys given all the chems. It fits most as an Assistant item, but should Assistants get something like this?

PKPenguin321 commented:
I'm going to tweak this and try to make the role restriction make more sense later.

MMMiracles commented:
n e r d r a g e

Robustin commented:
Running around like a turbohitler and doing backbreakers on randos feels like a much more assistant-themed item.

dannno commented:
sprite is shit but whatever
add growth serum to it tbh

Niknakflak commented:
im not sure why it's science jobs either.  Bane wasn't a scientist

Haevacht commented:
This is, fucking dumb.

PKPenguin321 commented:
For everybody asking why science I said that twice already
>the idea is that it was gonna be an augment for roboticists but I got lazy and didn't feel like fixing autoimplanters, and since it was less robotics-y I figured why not let all of science enjoy it

I'm going to tweak it to be more science-y or something though so it makes more sense don't worry

PKPenguin321 commented:
Not really feeling in a coding mood, yesterday was a rare thing where I spent all day working on this. I want to come back to this when I'm feeling up to it and not feel pressured to fix it quickly since a quick fix is usually a worse one.

You haven't seen the last of meeeee!
